RecoVertex/PrimaryVertexProducer/plugins/PrimaryVertexProducer.cc

```diff
@@ -263,8 +263,10 @@ void PrimaryVertexProducer::produce(edm::Event& iEvent, const edm::EventSetup& i
           if (v.isValid()) {
             auto err = v.positionError().matrix4D();
             err(3, 3) = vartime;
+            auto trkWeightMap3d = v.weightMap();  // copy the 3d-fit weights
             v = TransientVertex(
                 v.position(), meantime, err, v.originalTracks(), v.totalChiSquared(), v.degreesOfFreedom());
+            v.weightMap(trkWeightMap3d);
           }
         }
```

Comment:
now that this part is exactly the same as in the `useBeamConstraint` branch, please take it out of the if-else to avoid code replication (in retrospect, this PR wouldn't even be needed if it was already separateted)

Reply:
True. Done.